zdm/tests/mc_analysis.py

```diff
@@ -0,0 +1,845 @@
+#!/usr/bin/env python3
```

Comment:
OK this is HUGE! Is this Esan's work? Can you add a couple more comments at the top of the file describing what this is? (not that I do this enough... just do what I say, not what I do! ;-) )

Reply:
It is @esanmouli 's work.

Not sure all it entails! :)